### PR TITLE
Deploy to production button

### PR DESCRIPTION
**Pull request by jokabuyasina**
# Deploy to production button - One-Click Production Deployment

## Summary

Adds an intelligent deployment banner that automatically detects when new bundles are ready to deploy to production and provides a one-click deployment experience with visual feedback. The banner is admin-only and includes permission checks, confirmation dialogs, and celebratory animations.

## Problem

Deploying bundles to production currently requires 5 manual steps:
1. Navigate to Channels page
2. Find production channel
3. Click edit
4. Select latest bundle from dropdown
5. Confirm deployment

This is tedious for frequent deployments and increases human error risk.

## Solution

Smart banner that:
- ✅ Automatically appears when new bundle is available
- ✅ Shows only to admins (permission-checked)
- ✅ Deploys with single click + confirmation
- ✅ Provides instant visual feedback
- ✅ Celebrates with confetti animation on success

### Intelligent Detection
Banner appears only when **all** conditions are met:
- User has admin/super_admin/owner role
- Production channel exists
- Latest bundle differs from production channel version
- Bundle is valid (not "unknown" or "builtin")

### Permission System
- Read-only, upload, and write roles: **Cannot see banner**
- Admin, super_admin, owner roles: **Can see and deploy**
- Role fetched from organization store per app

### User Flow
1. Banner appears at top of app dashboard
2. User clicks "Deploy Now"
3. Confirmation dialog shows bundle details
4. User confirms → Production channel updated
5. Confetti animation plays
6. Success toast notification
7. Banner disappears after 1 second

## Technical Implementation

**Component**: `src/components/dashboard/DeploymentBanner.vue`
- Vue 3 Composition API with TypeScript
- Reactive state for loading/deploying states
- Computed properties for permission checks and visibility logic
- Supabase integration for channel updates
- Custom confetti animation system (50 particles, GPU-accelerated)

**Props**:
- `appId` (required): Application identifier

**Emitted Events**:
- `deployed`: Fired after successful deployment for parent refresh

**Error Handling**:
- Permission denied → Banner doesn't render
- Database errors → Toast error, user can retry
- Missing channel/bundle → Graceful degradation
- Concurrent deployments → Button disabled during operation

### New Files (3)
- `src/components/dashboard/DeploymentBanner.vue` - Main component (210 lines)
- `docs/DEPLOYMENT_BANNER.md` - Comprehensive documentation
- `PR_DESCRIPTION_DEPLOYMENT_BANNER.md` - Detailed PR description

### Integration Point
Component should be added to `src/pages/app/[package].vue`:
```vue
<DeploymentBanner :app-id="appId" @deployed="handleDeploymentComplete" />
```

### Manual Testing Steps
1. Log in as admin user (`admin@capgo.app`)
2. Navigate to app with production channel
3. Upload new bundle to staging
4. Verify banner appears with "Deploy Now" button
5. Click deploy → Confirm in dialog
6. Verify confetti animation + success toast
7. Verify banner disappears
8. Verify production channel updated in database

### Permission Testing
- Test with non-admin user → Banner should NOT appear
- Test with admin user → Banner should appear

### Edge Cases
- No production channel → Banner doesn't appear
- Already deployed latest → Banner doesn't appear
- Concurrent clicks → Only one deployment occurs

## Breaking Changes

None. This is a new feature with no impact on existing functionality.

## Deployment Notes

- No database migrations required
- No feature flags needed
- Feature immediately available to all admin users
- Apps without "production" channel gracefully skip banner

---

## Summary by CodeRabbit

* **New Features**
  * Admin-only deployment banner: detects new bundles, one-click deploy with confirmation, loading/deploying states, optimistic UI update, success toasts, confetti, and emits a deployed event for parents to refresh.
* **Integration**
  * Banner integrated into the app overview to trigger parent refresh after deployment.
* **Documentation**
  * Comprehensive docs added covering usage, security/permissions, workflow, i18n, troubleshooting, and testing guidance.
* **Localization**
  * Added English keys for the banner and fixed misspelled keys across multiple locales.
* **Chores**
  * Added typo-extension configuration.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

---

> [!NOTE]
> Adds a smart, permission-gated deployment control to streamline production releases.
> 
> - New `DeploymentBanner.vue` detects when `latestBundle.id !== productionChannel.version`, shows to admins only, and updates `channels.version` on confirm
> - Confirmation dialog, optimistic UI, success/error toasts, and confetti animation; emits `deployed` for parent refresh
> - Integrated into `src/pages/app/[package].vue` to display banner and refresh data post-deploy
> - Comprehensive docs: `docs/DEPLOYMENT_BANNER.md` and PR description file; i18n keys referenced
> - No breaking changes; minimal performance impact (a few Supabase queries)
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4c61d735debc6b247edb49600edb22dc951ce491. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/components/dashboard/DeploymentBanner.vue [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+/**
+ * Computed property that checks if the user has admin-level permissions.
+ * Only users with admin, super_admin, or owner roles can see and use the banner.
+ *
+ * @returns {boolean} True if user has admin permissions, false otherwise
+ */
+const hasAdminPermission = computed(() => {
+  return userRole.value ? organizationStore.hasPermissionsInRole(userRole.value, ['admin', 'super_admin']) : false
+})
```
[the remark this answers is not in the document]
Comment (riderx) [reply src/components/dashboard/DeploymentBanner.vue:80]: owner do not exist in our code
